test/Conversion/ExportVerilog/hw-lower-instance-choices.mlir

```diff
@@ -0,0 +1,38 @@
+// RUN: circt-opt %s -hw-lower-instance-choices --split-input-file -verify-diagnostics | FileCheck %s
+
+// CHECK: sv.macro.decl @__circt_choice_top_inst1
+// CHECK: sv.macro.decl @__circt_choice_top_inst2
+
+hw.module private @TargetA(in %a: i32, out b: i32) {
+  hw.output %a : i32
+}
+
+hw.module private @TargetB(in %a: i32, out b: i32) {
+  hw.output %a : i32
+}
+
+hw.module private @TargetDefault(in %a: i32, out b: i32) {
+  hw.output %a : i32
+}
+
+hw.module public @top(in %a: i32, out b: i32, out d: i32) {
+  // CHECK:               sv.ifdef  "__circt_choice_top_inst1" {
+  // CHECK-NEXT:          } else {
+  // CHECK-NEXT{LITERAL}:   sv.macro.def @__circt_choice_top_inst1 "{{0}}"([@TargetDefault])
+  // CHECK-NEXT:          }
```

Comment:
Aside: I haven't looked at this serialization format in a long time. This would be clearer if `sv.ifdef` serialized with an actual symbol and not a string.

Reply:
We have a bit of a problem here, we'd need macro-level expressions.